crates/bevy_ecs/src/system/mod.rs

```diff
@@ -138,9 +138,7 @@ mod tests {
         world.spawn().insert(A);
 
         system.initialize(&mut world);
-        for archetype in world.archetypes.iter() {
-            system.new_archetype(archetype);
-        }
+        system.update_archetypes(&world);
```

Comment:
This test doesn't really check if the archetypes are updated correctly.  Since this new function is part of the public api that we expect to be used, should we be adding more tests to check correctness?